modin/pandas/test/test_concat.py

```diff
@@ -23,11 +23,17 @@
     generate_multiindex_dfs,
     generate_none_dfs,
     create_test_dfs,
+    default_to_pandas_ignore_string,
 )
-from modin.config import NPartitions
+from modin.config import NPartitions, StorageFormat
 
 NPartitions.put(4)
 
+pytestmark = pytest.mark.filterwarnings(default_to_pandas_ignore_string)
+
+# Initialize env for storage format detection in @pytest.mark.*
+pd.DataFrame()
```

Comment:
Are you sure this is necessary? `StorageFormat` should get the right value for `@pytest.mark.skipif(` without explicit initialization.

Reply:
If I comment out this line, I'm getting 1152 tests skipped in the HDK environment. StorageFormat.get() returns "Pandas".

Reply:
I can't reproduce this problem. Moreover, we already have a place in the tests that checks the backend, but at the same time, the dataframe is not explicitly initialized in that file: https://github.com/modin-project/modin/blob/8df0030e1c1cdde73591fefe1d2fd7a164312b3a/modin/pandas/test/test_io.py#L70

Let's delete it and see how the CI goes.

Reply:
> I can't reproduce this problem.

Have you checked this env - requirements/env_hdk.yml ?

Reply:
No, I use the following code:
```python
import os
os.environ["MODIN_EXPERIMENTAL"] = "True"
os.environ["MODIN_ENGINE"] = "native"
os.environ["MODIN_STORAGE_FORMAT"] = "hdk"

from modin.config import StorageFormat
print(StorageFormat.get())
```

Reply:
It works if you initialise the environment explicitly.
Try to remove this line - pd.DataFrame() and run the test in this - requirements/env_hdk.yml environment, without setting the environment variables:

```
mamba create -f requirements/env_hdk.yml
mamba activate modin_on_hdk
python -m pytest -n 2 modin/pandas/test/test_concat.py
```

Reply:
Now it is clear. However, although Modin can determine the engine and backend based on installed dependencies, the design of the tests does not imply that the environment is not configured.

Reply:
@AndreyPavlenko for this behavior, we need a centralized solution (for all files with tests), I will be glad if you create an issue for this. Good suggestion, thanks!

Reply:
I absolutely agree we need a solution. The solution, implemented in this fix, is just an ugly workaround.

Reply:
> I will be glad if you create an issue for this.

I've created the issue - https://github.com/modin-project/modin/issues/5712